torch/quantization/_numeric_suite.py

```diff
@@ -39,3 +42,139 @@ def compare_weights(float_dict, quantized_dict):
             weight_dict[key]["float"] = float_dict[match_key]
             weight_dict[key]["quantized"] = quantized_dict[key]
     return weight_dict
+
+
+def _get_observer_dict(mod, target_dict, observer_name, prefix=""):
+    r"""Traverse the modules and save all observers into dict.
+    This is mainly used for quantization accuracy debug
+    Args:
+        mod: the top module we want to save all observers
+        prefix: the prefix for the current module
+        observer_name: the name of observer we want to get, "logger" is used
+            to do the module level comparison between quantized module and its
+            matching float shadow module, and "activation_post_process" is
+            used to compare the module outputs between float and quantized
+            models
+        target_dict: the dictionary used to save all the observers
+    """
+
+    def get_prefix(prefix):
+        return prefix if prefix == "" else prefix + "."
+
+    if hasattr(mod, observer_name):
+        if observer_name == "logger":
+            target_dict[get_prefix(prefix) + "stats"] = mod.logger.stats
+        elif observer_name == "activation_post_process":
+            target_dict[
+                get_prefix(prefix) + "stats"
+            ] = mod.activation_post_process.tensor_val
+
+    for name, child in mod.named_children():
+        module_prefix = get_prefix(prefix) + name if prefix else name
+        _get_observer_dict(child, target_dict, observer_name, module_prefix)
+
+
+class Logger(nn.Module):
```

Comment:
Create a base class that only has a stats dict but no forward method and inherit logger from that class.